mypy/erasetype.py

```diff
@@ -140,9 +140,12 @@ class LastKnownValueEraser(TypeTranslator):
     Instance types."""
 
     def visit_instance(self, t: Instance) -> Type:
-        if t.last_known_value:
-            return t.copy_modified(last_known_value=None)
-        return t
+        if not t.last_known_value and not t.args:
+            return t
+        return t.copy_modified(
+            args=[a.accept(self) for a in t.args],
+            last_known_value=None,
+        )
```

Comment:
Good catch! Did you add a test for this?

Reply:
Good point -- I initially didn't since we already had several tests unrelated to literal types that were breaking because of this.

But I think it'd be good to exercise this more directly, so I added a test to `check-literals.test`.